### PR TITLE
Switch to dune xt

### PR DESCRIPTION
**Pull request by tobiasleibner**
First changes to switch to dune-xt #173 , mainly adjusted include paths. Tests and headerchecks pass, but there are probably some additional changes needed for CI.

### CODE REVIEW

**Comment by tobiasleibner** [comment]
This mostly works, see [build log](https://zivgitlab.uni-muenster.de/ag-ohlberger/dune-community/dune-gdt/-/jobs/194071). The remaining issue is that there is no ``begin_testcases()`` anymore in ``dune-xt``, but only ``add_subdir_tests``. I worked around this by calling ``add_subdir_tests(test)`` in directory ``dune/gdt`` instead of calling ``begin_testcases`` in ``dune/gdt/test``. However, now the test binaries are created in directory ``dune/gdt`` as well. There seems to be a lot of ``dune-xt`` specific stuff in ``add_subdir_tests`` now, so the easiest fix would be to just copy ``add_subdir_tests`` to something like ``add_gdt_tests`` and adjust it accordingly. However, that would result in a lot of code duplication. Any ideas, @renefritze?

**Comment by tobiasleibner** [comment]
Curiously, the [test](https://dev.azure.com/dune-community/dune-xt-data/_build/results?buildId=1048&view=logs&j=8f867060-48de-572a-2d06-2f187b070db0) that fails in dune-xt-data seems to work here.

**Comment by ftalbrecht** [comment]
Since we already have all tests in subdirs, cant we just do the same thing as in `dune-xt`, i.e. add all subdirs?

**Comment by tobiasleibner** [comment]
Yep, that should work... didn't think of that. Testing now.

**Comment by tobiasleibner** [comment]
Nice, CI is green, except that the PR build failed because it used the ``master`` branch in ``dune-gdt-super`` instead of the ``switch_to_dune_xt`` branch. I merged that branch into ``master``, once the ``dune-gdt-super`` builds are ready I will restart the build here. If there are no objections, I will merge this PR once both builds are passing.

**Comment by ftalbrecht** [comment]
No objections!

File: dune.module [hunk 1]
```diff
@@ -16,5 +16,5 @@
 Module: dune-gdt
 Version: 18.11-dev
 Maintainer: felix.schindler@wwu.de
-Depends: dune-localfunctions dune-xt-grid (>= 18.11) dune-xt-la (>= 18.11) dune-xt-functions (>= 18.11)
+Depends: dune-localfunctions dune-xt (>= 18.11)
 Suggests: dune-fem (>= 1.3)
```
Comment (ftalbrecht) [inline dune.module]: Just a side note: do we even have anything depending on dune-fem left in dune-gdt? I could not find a single include. If not, we should also remove the suggestion in the `dune.module`.

Comment (tobiasleibner) [reply dune.module]: Probably not, I will remove the suggestion.

**Pull request merged**